src/plugins/workspace/public/workspace_client.ts

```diff
@@ -117,13 +117,21 @@ export class WorkspaceClient implements IWorkspaceClient {
         perPage: 999,
         permissionModes: [WorkspacePermissionMode.LibraryWrite],
       });
-      if (resultWithWritePermission?.success) {
+      const resultWithOwnerPermission = await this.list({
+        perPage: 999,
+        permissionModes: [WorkspacePermissionMode.Write],
+      });
```

Comment:
can we combine it with line 118?

Reply:
No, we need get both owner list and read/write list to set readonly and owner attribute of workspace object.